PyMca5/PyMcaIO/HDF5Stack1D.py

```
@@ -214,8 +214,14 @@ def loadFileList(self, filelist, selection, scanlist=None):
                     scanlist.append("")
         else:
             try:
-                number, order = [int(x) for x in scanlist[0].split(".")]
-                JUST_KEYS = True
+                # the ESRF uses "1.1" notation for the scans so this is ambiguous because
+                # one does not know if the selection is based on scan name or scan order
+                if scanlist[0] in entryNames:
+                    # assumed the selection is based on scan name
+                    JUST_KEYS = False
+                else:
+                    number, order = [int(x) for x in scanlist[0].split(".")]
+                    JUST_KEYS = True
             except:
```

Bare exceptions like this will capture all `BaseExceptions` like `KeyboardInterrupt` and `SystemExit`.

---

In 17 years of code maintenance I have seen exceptions changing type and therefore not being caught.

In situations where multiple exceptions can happen, I would prefer to explicitly catch `KeyboardInterrupt` and `SystemExit` to re-raise them rather than missing any possible exception.

You see that philosophy in PyMca when importing optional dependencies. Instead of checking for ImportError, I prefer to catch any exception to avoid situations like https://github.com/silx-kit/silx/issues/3731

---

Are there any other base exceptions than `KeyboardInterrupt` and `SystemExit` that you would like to get propagated?

---

A bare except

```
try:
    ...
except:
    ...
```

is equivalent to

```
try:
    ...
except BaseException:
    ...
```

If you want to capture all non-lethal exceptions you can do

```
try:
    ...
except Exception:
    ...
```

This ensures you don't capture exceptions that should never be _absorbed_ (i.e. if you capture them you should ensure a re-raise).

---

For this reason any linter like `flake8` warns against bare excepts.

---

OK. I will change `except:` to `except Exception:`

---

#980 addresses it.